app/src/main/java/com/arctouch/codechallenge/repository/implementation/GenresRepositoryImplementation.java

```diff
@@ -0,0 +1,43 @@
+package com.arctouch.codechallenge.repository.implementation;
+
+import com.arctouch.codechallenge.api.TmdbApi;
+import com.arctouch.codechallenge.api.TmdbApiClient;
+import com.arctouch.codechallenge.callback.OnGetGenres;
+import com.arctouch.codechallenge.data.Cache;
+import com.arctouch.codechallenge.entity.GenreResponse;
+import com.arctouch.codechallenge.repository.GenresRepository;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class GenresRepositoryImplementation implements GenresRepository {
+
+    private TmdbApi service;
+
+    public GenresRepositoryImplementation() {
+        service = TmdbApiClient.createService(TmdbApi.class);
+    }
+
+    @Override
+    public void getGenres(OnGetGenres onGetGenres) {
+        service.genres().enqueue(new Callback<GenreResponse>() {
```

Comment:
Why do you prefer to Callbacks instead of Rx?

Reply:
That was just a matter of strategy. Based on the short time to deliver, I went with what was most comfortable to me, despite not being the most sophisticated approach. Could have gone with Rx, though. Its nice!